core/key.go

```diff
@@ -106,6 +116,13 @@ type CollectionSchemaVersionKey struct {
 
 var _ Key = (*CollectionSchemaVersionKey)(nil)
 
+type CollectionIndexKey struct {
```

Comment:
I actually don't think it should be string names here as far as I know, thoughts @islamaliev ?

Reply:
what is explained in SIP is related to keys for docs' indexed field values. And indeed there we use IDs:
```go
type IndexDatastoreKey struct {
	CollectionID string
	IndexID	     string
	FieldValues  []string
}
```
It's important there to keep the index as short as possible for better fetching performance.
The one in question in turn represent a key that points for an index description which is not supposed to be used as often and doesn't impose any requirements on traits like soft-ability.
On the other hand it will save us few fetches when, for example, a user requests to list all indexes. In this case we would need just to through the keys.
What do you think?

Reply:
> On the other hand it will save us few fetches when, for example, a user requests to list all indexes. In this case we would need just to through the keys.

IMO that is far less importance performance-wise than whatever work we need to do on a normal fetch/query using an index.

> It's important there to keep the index as short as possible for better fetching performance.

This to me suggests that an int (or smaller) would be much better here, for both collectionID and indexID no? Particularly for `IndexDatastoreKey`.

Reply:
you are right, it's not the important. I guess I chose this approach because it was a tiny bit easier to implement.
But I still didn't hear any strong objections against `CollectionName` vs `CollectionID`.
Regarding `IndexDatastoreKey`, it's actually supposed to be given int, but converted to string. I saw other examples (other keys) that do it this way. Wasn't sure why myself.

Reply:
> it's actually supposed to be given int, but converted to string. I saw other examples

Are you thinking about `FieldID`?  `FieldID` is like that as it is/can-be a string, as sometimes it is assigned `C` for composites. I cant think of any other props that do that off the top of my head.

Reply:
Ah, I see `CollectionKey` is really just an index of the collection by collection name, that is it's only usage as far as I can see.